### PR TITLE
Add the ability to ignore lines from coverage depending on the comment

### PR DESCRIPTION
**Pull request by pasindud**
Use // coverage:ignore-line to ignore one line.
Use // coverage:ignore-start and // coverage:ignore-end to ignore range of lines inclusive.
Use // coverage:ignore-file to ignore the whole file.

### CODE REVIEW

File: bin/format_coverage.dart [hunk 1]
```diff
@@ -39,10 +40,16 @@ Future<Null> main(List<String> arguments) async {
     print('  package-root: ${env.pkgRoot}');
     print('  package-spec: ${env.packagesPath}');
     print('  report-on: ${env.reportOn}');
+    print('  check-ignore: ${env.checkIgnore}');
   }
 
   final clock = Stopwatch()..start();
-  final hitmap = await parseCoverage(files, env.workers);
+  print(env.checkIgnore);
```
Comment (grouma) [inline bin/format_coverage.dart:47]: Should this be removed?

**Comment by pasindud** [comment]
Should we add a entry to README also ?

**Comment by grouma** [comment]
> Should we add a entry to README also about this ?

Probably a good idea. Otherwise LGTM.

**Comment by Elias8** [comment]
Is this available on the current stable which is `1.22.5`?

**Comment by grouma** [comment]
This should be available in stable as it depends on version 14.1: https://github.com/flutter/flutter/blob/78910062997c3a836feee883712c241a5fd22983/packages/flutter_tools/pubspec.yaml#L17

**Comment by lukepighetti** [comment]
I'm on Flutter `Channel dev, 1.26.0-8.0.pre` and have `// coverage:ignore-file` at the top of a file and it's still showing up in code coverage reports. Am I missing a configuration step?

**Comment by grouma** [comment]
@lukepighetti I tried to reproduce your issue and I'm unable to do so. I simply ran `flutter test --coverage` on the `hello_world` example within the Flutter repo. I can add and ignore files as expected.

**Pull request merged**